cloudinit/config/cc_ca_certs.py

```diff
@@ -198,26 +246,37 @@ def handle(
     # If there is a remove_defaults option set to true, remove the system
     # default trusted CA certs first.
     if "remove-defaults" in ca_cert_cfg:
-        log.warning(
+        LOG.warning(
             "DEPRECATION: key 'ca-certs.remove-defaults' is now deprecated."
             " Use 'ca_certs.remove_defaults' instead."
         )
-        if ca_cert_cfg.get("remove-defaults", False):
-            log.debug("Removing default certificates")
-            remove_default_ca_certs(cloud.distro.name, distro_cfg)
+        if cloud.distro.name in ["fedora", "rhel"]:
+            LOG.error(
+                "Disabling default certificates is not supported for"
+                " this distro."
+            )
+        elif ca_cert_cfg.get("remove-defaults", False):
+            LOG.debug("Disabling default certificates")
+            disable_default_ca_certs(cloud.distro.name, distro_cfg)
     elif ca_cert_cfg.get("remove_defaults", False):
-        log.debug("Removing default certificates")
-        remove_default_ca_certs(cloud.distro.name, distro_cfg)
+        if cloud.distro.name in ["fedora", "rhel"]:
+            LOG.error(
```

Comment:
So basically fedora and rhel won't support this option? I didn't understand why

Reply:
ah ok now I got it. I would wait for https://bugzilla.redhat.com/show_bug.cgi?id=2163554 opened by @holmanb before disabling it to fedora and rhel. I will also ask if someone from our side can help fixing that in RHEL.

Reply:
@esposem unfortunately the response in that Redhat bug referred to "local" certificates which was not what the question actually as regarding - it was regarding what is the official/"proper" way in RHEL/Fedora to disable/delete the system CA certificates.

Anyway as @holmanb suggested I have reintroduced the previous behaviour - though only for the /etc/pki/ca-trust/ directory and not for the /usr/share/pki/ca-trust-source/ directory.

Reply:
I also tried to ping the person internally, no answer so far. I guess if this is the only blocker of this PR, just go ahead, we'll introduce the proper behaviour later. If I find something I will let you know.

Reply:
Removed Fedora and RHEL related changes from this PR.